custom_components/nordpool/test_parser.py

```diff
@@ -1,42 +1,230 @@
+import asyncio
 import logging
-import sys  # Make sure plexapi is in the systempath
 from collections import defaultdict
 from datetime import date, datetime, timedelta
 from operator import itemgetter
-from os.path import abspath, dirname
-from pprint import pprint
 
-import requests
+import aiohttp
+from aiozoneinfo import async_get_time_zone
 # https://repl.it/repls/WildImpishMass
 from dateutil import tz
 from dateutil.parser import parse as parse_dt
-from nordpool import elspot
+from nordpool.base import CurrencyMismatch
+from nordpool.elspot import Prices
 
+_LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+
+INVALID_VALUES = frozenset((None, float("inf")))
+
+
+class InvalidValueException(ValueError):
+    pass
+
+
+
+class AioPrices(Prices):
+    """Interface"""
 
-class PP(elspot.Prices):
-    def __init__(self, currency):
+    def __init__(self, currency, client, timeezone=None):
         super().__init__(currency)
-        self.API_URL_CURRENCY = "https://www.nordpoolgroup.com/api/marketdata/page/%s"
+        self.client = client
+        self.timeezone = timeezone
+        (self.HOURLY, self.DAILY, self.WEEKLY, self.MONTHLY, self.YEARLY) = ("DayAheadPrices", "AggregatePrices",
+                                                                             "AggregatePrices", "AggregatePrices",
+                                                                             "AggregatePrices")
+        self.API_URL = "https://dataportal-api.nordpoolgroup.com/api/%s"
 
-    def _fetch_json(self, data_type, end_date=None):
-        ''' Fetch JSON from API '''
+    async def _io(self, url, **kwargs):
+
+        resp = await self.client.get(url, params=kwargs, headers={
+            'Origin': 'https://data.nordpoolgroup.com'
+        })
+        _LOGGER.debug("requested %s %s", resp.url, kwargs)
+
+        return await resp.json()
+
+    def _parse_json(self, data, areas=[]):
```

Comment:
Change area to None and the check if areas is None. We do not want mutable keyword arguments

Reply:
What is test_parser used for?